app/repositories/document_repository.rb

```diff
@@ -2,7 +2,34 @@
 
 class DocumentRepository
   include Elasticsearch::Persistence::Repository
+  include Elasticsearch::Persistence::Repository::DSL
+
+  extend NamespacedIndex
 
   klass Document
   client ES.client
+
+  def serialize(document)
+    document_hash = ActiveSupport::HashWithIndifferentAccess.new(super)
+    Serde.serialize_hash(document_hash, document_hash[:language])
+  end
+
+  def deserialize(hash)
+    doc_hash = hash['_source']
+    deserialized_hash = Serde.deserialize_hash(doc_hash,
+                                               doc_hash['language'])
+
+    document = Document.new deserialized_hash
+    document.instance_variable_set :@_id, hash['_id']
+    document.instance_variable_set :@_index, hash['_index']
+    document.instance_variable_set :@_type, hash['_type']
+    document.instance_variable_set :@_version, hash['_version']
+
+    document.instance_variable_set :@hit, Hashie::Mash.new(
+      hash.except('_index', '_type', '_id', '_version', '_source')
+    )
+
+    document.instance_variable_set(:@persisted, true)
+    document
+  end
```

Comment:
These serialization  methods have been moved largely unchanged from the original `Document` class. I was tempted to clean up the `deserialize` method, but decided to avoid falling into another refactoring black hole.